EXILED/Exiled.API/Features/Toys/Speaker.cs

```diff
@@ -108,5 +141,144 @@ public static Speaker Create(Vector3? position, Vector3? rotation, Vector3? scal
+        /// </summary>
+        /// <param name="path">The file path of the audio file.</param>
+        /// <param name="volume">The desired playback volume. (0 to <see cref="float"/>) max limit.</param>
+        /// <param name="minDistance">The minimum distance at which the audio is audible.</param>
+        /// <param name="maxDistance">The maximum distance at which the audio is audible.</param>
+        /// <param name="destroyAfter">Whether the Speaker gets destroyed after it's done playing.</param>
+        public void Play(string path, float volume, float minDistance, float maxDistance, bool destroyAfter)
+        {
+            if (IsPlaying)
+                Stop();
+
+            if (!File.Exists(path))
+            {
+                Log.Warn($"Tried playing audio at {path} but no file was found.");
+                return;
+                            }
+
+            Volume = volume;
+            MinDistance = minDistance;
+            MaxDistance = maxDistance;
+
+            IsPlaying = true;
+            Timing.RunCoroutine(PlaybackRoutine(path, destroyAfter));
+        }
+
+        /// <summary>
+        /// Stops the current playback.
+        /// </summary>
+        public void Stop()
+        {
+            stopPlayback = true;
+            IsPlaying = false;
+        }
+
+        private IEnumerator<float> PlaybackRoutine(string filePath, bool destroyAfter)
+        {
+            stopPlayback = false;
+
+            string fileExtension = Path.GetExtension(filePath).ToLower();
+            Log.Debug($"Detected file: {filePath}, Extension: {fileExtension}");
+
+            const int sampleRate = 48000; // Enforce 48kHz
+            const int channels = 1; // Enforce mono audio
+            const int frameSize = 480; // Frame size for 10ms of audio at 48kHz
+
+            Queue<float> streamBuffer = new();
+            float[] readBuffer = new float[frameSize * 4];
+            float[] sendBuffer = new float[frameSize];
+            byte[] encodedBuffer = new byte[512];
+            OpusEncoder encoder = new(VoiceChat.Codec.Enums.OpusApplicationType.Voip);
+
+            float playbackInterval = frameSize / (float)sampleRate;
+            float nextPlaybackTime = Timing.LocalTime;
+
+            if (fileExtension != ".ogg")
+            {
+                Log.Error($"Unsupported file format: {fileExtension}");
+                yield break;
+            }
+
+            using VorbisReader vorbisReader = new(filePath);
+
+            if (vorbisReader.SampleRate != sampleRate || vorbisReader.Channels != channels)
+            {
+                Log.Error($"Invalid OGG file. Expected {sampleRate / 1000}kHz mono, got {vorbisReader.SampleRate / 1000}kHz {vorbisReader.Channels} channel(s).");
+                yield break;
+            }
+
+            Log.Debug($"Playing OGG file with Sample Rate: {sampleRate}, Channels: {channels}");
+
+            while ((streamBuffer.Count < frameSize * 2 && !stopPlayback) || Round.IsEnded)
```

Comment:
Why round.IsEnded here ?

Reply:
It's supposed to end the playback if the round ends but I'm not sure if it even works

Reply:
Don't understand why tho this is seen like we can remove it, but idk why someone wants to have audio after the round is ended its most likely that the scene will be reset but not the coroutine so thats why

Reply:
If you think otherwise just let me know and a memory for you, we can see the edits you changed the message to not make it like we made it obvious but we didn't it was a choice by both of us. So next time before commenting please choose a nicer language as i think detailing the work put as "Not Cared" is not describing correctly what we did.

Reply:
And if someone else thinks it like him, let me know because my position can change and i can change it, i don't see now changing it.

Reply:
Ok after this missunderstanding, yes i now understand what you meant and im going to change it